runtime/client_http_request.go

```diff
@@ -82,7 +85,7 @@ func (req *ClientHTTPRequest) start(
 
 // WriteJSON will send a json http request out.
 func (req *ClientHTTPRequest) WriteJSON(
```

Comment:
Maybe change method name to  WriteJSONReq or WriteHTTPJSON, to make it clear you're writing the whole request with headers and not just the json body.